tfjs-core/src/io/io_utils.ts

```diff
@@ -330,10 +336,15 @@ export function base64StringToArrayBuffer(str: string): ArrayBuffer {
 /**
  * Concatenate a number of ArrayBuffers into one.
  *
- * @param buffers A number of array buffers to concatenate.
+ * @param buffers An array of ArrayBuffers to concatenate, or a single
+ *     ArrayBuffer.
  * @returns Result of concatenating `buffers` in order.
  */
-export function concatenateArrayBuffers(buffers: ArrayBuffer[]): ArrayBuffer {
+export function concatenateArrayBuffers(buffers: ArrayBuffer[]
```

Comment:
Can we move this to be part of `CompositeArrayBuffer`? like static method `CompositeArrayBuffer.join(buffers: ArrayBuffer[])` or through public method `new CompositeArrayBuffer(buffers).toArrayBuffer()`, which makes it easier to bridge CompositeArrayBuffer with native ArrayBuffer and pass CompositeArrayBuffer around in the future if needed.

Reply:
I was considering that, and my original implementation actually used `new CompositeArrayBuffer(buffers).slice()`, but I removed it in favor of `concatenateArrayBuffers` because of an issue with the types in tfjs-converter tests ([here was my fix for it in the `spy_ops.ts` file](https://github.com/tensorflow/tfjs/pull/7609/commits/cf2493e67a24fd1f0e5496f409e84a50d3c453a7#diff-02ae9d1c277302d454a7f39b9dd0793fbaedcc497ce7262c0c3b6fe2c0a07b77), but it's a bit hacky).

I'm fine with using the converter spy_ops.ts fix if it'll make the core implementation cleaner. What do you think?

Reply:
I took a look at the usage of `spyOnAllFunctions` in tests, and I think the test is something we should fix. I think a hacky way like what you did is probably fine.

In general, instead of automatically replace everything with spy using `spyOnAllFunctions`, I think we should explicitly create an ioSpy object which only contains the function we want to spy, so that we can make the test more controllable and reliable. There are some stuffs exported in io apparently should not be spied, like `getWeightSpecs`, which is a io helper function instead of a function to do io.

Reply:
@chunnienc I've replaced `concatenateArrayBuffers` with `CompositeArrayBuffer.join` in tfjs-core and deprecated `concatenateArrayBuffers`. We can't replace it in other packages yet because that would introduce a breaking change. Downstream packages could not be used with an earlier version of tfjs-core that does not implement `CompositeArrayBuffer` (see #7273 for an example of why this is important). We can apply this change to all the packages in the next major release.

Reply:
Actually, it's fine to use it in tests, since users will never run those. I'll swap `concatenateArrayBuffers` for `CompositeArrayBuffer.join` in the test files.